x/gamm/pool-models/stableswap/msgs.go

```diff
@@ -51,6 +53,11 @@ func (msg MsgCreateStableswapPool) ValidateBasic() error {
 	} else if len(msg.InitialPoolLiquidity) > 2 {
 		return types.ErrTooManyPoolAssets
 	}
+	if len(msg.ScalingFactors) != 2 {
+		return types.ErrInvalidScalingFactors
+	} else if msg.ScalingFactors[0] == 0 || msg.ScalingFactors[1] == 0 {
+		return types.ErrInvalidScalingFactors
+	}
```

Comment:
```suggestion
	// valid scaling factor lengths are 0, or one factor for each asset
	if len(msg.ScalingFactors) != 0 && len(msg.ScalingFactors) != len(msg.InitialPoolLiquidity) {
		return types.ErrInvalidScalingFactors
	}
```

Reply:
Shoudn't we also check for zero scaling factor

Reply:
0 scaling factor should get inferred to just be all 1's